### ISSUE TITLE
dbt - sequence item 0: expected str instance, Nonetype found when linting an entire dbt project

### ISSUE THREAD

**Issue by fathom-parth** [sqlfluff/sqlfluff#4449]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have a dbt lineage graph that looks like:
source table -> staging view -> intermediate ephemeral 1 -> intermediate ephemeral 2 -> intermediate ephemeral 3 -> mart table

When I had only 1 intermediate ephemeral model, I didn't hit any errors but after adding the 2 additional ephemeral modells I get dbt compilation errors only when running sqlfluff. dbt commands still run fine.

Additionally, if I manually point sqlfluff to my mart table or ephemeral models to lint, it lints correctly without errors.

### Expected Behaviour

I expect sqlfluff to run correctly and not throw errors when linting.

### Observed Behaviour

```shell
❯ sqlfluff lint .
=== [dbt templater] Sorting Nodes...                                                                                                              
file models/intermediate/training/int_1.sql:   0%|                               | 0/9 [00:00<?, ?it/s]22:00:56  Unable to do partial parsing because config vars, config profile, or config target have changed
=== [dbt templater] Compiling dbt project...                                                                                                      
=== [dbt templater] Project Compiled.                                                                                                             
ERROR      Fatal dbt compilation error on .../dbt/models/intermediate/training/int_2.sql. This occurs most often during incorrect sorting of ephemeral models before linting. Please report this error on github at https://github.com/sqlfluff/sqlfluff/issues, including both the raw and compiled sql for the model affected. 
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/sqlfluff_templater_dbt/templater.py", line 484, in _unsafe_process
    manifest=self.dbt_manifest,
  File "/usr/local/lib/python3.7/site-packages/dbt/compilation.py", line 531, in compile_node
    node, _ = self._recursively_prepend_ctes(node, manifest, extra_context)
  File "/usr/local/lib/python3.7/site-packages/dbt/compilation.py", line 319, in _recursively_prepend_ctes
    prepended_ctes,
  File "/usr/local/lib/python3.7/site-packages/dbt/compilation.py", line 241, in _inject_ctes_into_sql
    token = sqlparse.sql.Token(sqlparse.tokens.Keyword, ", ".join(c.sql for c in ctes))
TypeError: sequence item 0: expected str instance, NoneType found
WARNING    Skipped file .../dbt/models/intermediate/training/int_3.sql because dbt raised a fatal exception during compilation: sequence item 0: expected str instance, NoneType found 
ERROR      Fatal dbt compilation error on .../dbt/models/marts/training/mart.sql. This occurs most often during incorrect sorting of ephemeral models before linting. Please report this error on github at https://github.com/sqlfluff/sqlfluff/issues, including both the raw and compiled sql for the model affected. 
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/sqlfluff_templater_dbt/templater.py", line 484, in _unsafe_process
    manifest=self.dbt_manifest,
  File "/usr/local/lib/python3.7/site-packages/dbt/compilation.py", line 531, in compile_node
    node, _ = self._recursively_prepend_ctes(node, manifest, extra_context)
  File "/usr/local/lib/python3.7/site-packages/dbt/compilation.py", line 319, in _recursively_prepend_ctes
    prepended_ctes,
  File "/usr/local/lib/python3.7/site-packages/dbt/compilation.py", line 241, in _inject_ctes_into_sql
    token = sqlparse.sql.Token(sqlparse.tokens.Keyword, ", ".join(c.sql for c in ctes))
TypeError: sequence item 0: expected str instance, NoneType found
WARNING    Skipped file .../dbt/models/marts/training/mart.sql because dbt raised a fatal exception during compilation: sequence item 0: expected str instance, NoneType found 
file models/staging/1.sql: 100%|████████████████████████████████████████████████████████████| 9/9 [01:00<00:00, 10.78s/it]All Finished 📜 🎉!  
```

### How to reproduce

You should be able to reproduce in a dbt environment where you have 3 ephemeral models being rendered into a single mart.

### Dialect

BigQuery

### Version

sqlfluff==2.0.0a5
python==3.7.16

### Configuration

.sqlfluff
```
[sqlfluff]
templater = dbt
dialect = bigquery
max_line_length = 88

[sqlfluff:templater:dbt]
profile = diseaseTools
target = dev

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:identation]
tab_space_size = 4
indent_unit = space

[sqlfluff:rules:L038]
select_clause_trailing_comma = require
```

.sqlfluffignore
```
# ignore linting compiled sql
target/

# dbt <1.0.0
dbt_modules/

# dbt >=1.0.0
dbt_packages/

# ignore linting macros (tests are just fancy macros)
macros/
tests/
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by fathom-parth** [comment]
Using 2.0.0a5 due to:
https://github.com/sqlfluff/sqlfluff/issues/4392

This also happens on 2.0.0a4

**Comment by fathom-parth** [comment]
It's sadly hard for me to provide the raw sql due to the query being sensitive hence why I didn't.

**Comment by fathom-parth** [comment]
Possibly related to:
https://github.com/dbt-labs/dbt-core/issues/6885

**Comment by fathom-parth** [comment]
Fixed in new dbt version:
```
❯ dbt --version
Core:
  - installed: 1.4.5
  - latest:    1.4.5 - Up to date!

Plugins:
  - bigquery: 1.4.2 - Up to date!
```
